### PR TITLE
RL-192 REST Service for batch create Relationships

### PR DESCRIPTION
**Pull request by zemlyanoi**
## Description
REST Service for batch create Relationships

## Jira Issue link
https://osi-cwds.atlassian.net/browse/RL-192

## Tests
- [x] I have included unit tests 
- [ ] I have included functional tests 
- [x] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the google code style of this project.
- [ ] I have ran all tests for the project.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check SonarQube after the build completes, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/rest/resources/ScreeningRelationshipResource.java [hunk 1]
```diff
@@ -152,11 +152,11 @@ public Response get(@PathParam("id") @ApiParam(required = true,
           @ApiResponse(code = HttpStatus.SC_NOT_FOUND, message = "Not found")})
   @ApiOperation(value = "Create Screening Relationships", code = HttpStatus.SC_CREATED,
       response = ScreeningRelationshipBase[].class)
-  public Response batchCreate(
-      @Valid @ApiParam(required = true) ScreeningRelationshipBase[] screeningRelationships) {
-    return Response.ok()
-        .entity(relationshipFacade.createRelationships(Arrays.asList(screeningRelationships)))
-        .build();
+  public Response batchCreate(@Valid @ApiParam(
```
Comment (tparkerosi) [inline src/main/java/gov/ca/cwds/rest/resources/ScreeningRelationshipResource.java]: How is Batch RESTful? This is doesn't follow the noun verb conventions of rest.

File: src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacade.java [hunk 2]
```diff
@@ -92,22 +91,15 @@ private void initSystemCodes() {
...
-          public String error = "Test Error";
-        };
-        responses.add(relationshipWithError);
-      } else {
-        ScreeningRelationship relationshipWithError = new ScreeningRelationship(
-            relationships.get(0));
-        relationshipWithError.setId(i + "");
-        responses.add(relationshipWithError);
+    relationships.forEach(relationship -> {
+      try {
```
Comment (tparkerosi) [inline src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacade.java]: what throws an exception here? is it the add or the create? My first thought is not to rerun the createRelationship when we have an exception, but to use an existing object so we arn't duplicating that code. But it looks to me that the create throws the exception, wouldn't it just rethrow it inside the catch block? Am I missing something?

File: src/test/java/gov/ca/cwds/rest/resources/relationship/ScreeningRelationshipResourceIRT.java [hunk 3]
```diff
@@ -85,6 +98,108 @@ public void testCreateRelationships() throws IOException, JSONException {
...
         doPostCall(SCREENING_RELATIONSHIPS_BATCH, request);
+    System.out.println(getStringResponse(response));
     assertEquals(HttpStatus.SC_OK, response.getStatus());
   }
+
+  @Test
+  public void createRelationships_createOneRelationship() throws IOException {
+    ScreeningRelationshipBase[] relationshipBases = {
+        getOneRelationshipForCreate("2033", "2034", 185, "Y")};
+    valedateCreateRelationships(relationshipBases);
```
Comment (tparkerosi) [inline src/test/java/gov/ca/cwds/rest/resources/relationship/ScreeningRelationshipResourceIRT.java]: minor spelling error

File: src/test/java/gov/ca/cwds/rest/resources/relationship/ScreeningRelationshipResourceIRT.java [hunk 4]
```diff
@@ -85,6 +98,108 @@ public void testCreateRelationships() throws IOException, JSONException {
...
+    Arrays.asList(relationshipBases).forEach(relationshipBase -> {
+      Optional<ScreeningRelationship> optional = actualResponse.stream().filter(
+          relationship -> relationship.getClientId().equals(relationshipBase.getClientId())
+              && relationship.getRelativeId().equals(relationshipBase.getRelativeId())).findFirst();
+      if (optional.isPresent()) {
+        validateResponseForCreatedRelationships(relationshipBase, optional.get());
+      }
+    });
+  }
+
```
Comment (tparkerosi) [inline src/test/java/gov/ca/cwds/rest/resources/relationship/ScreeningRelationshipResourceIRT.java]: I like this! Lots of people would repeat this code all over.

File: src/main/java/gov/ca/cwds/rest/resources/ScreeningRelationshipResource.java (hunk 1 above)
Comment (tparkerosi) [reply src/main/java/gov/ca/cwds/rest/resources/ScreeningRelationshipResource.java]: I know this wasn't in this request, but I'm wondering why. We should be creating restful endpoints. We are getting more and more that are deviating from the REST conventions and I would like to stop it.

File: src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacade.java (hunk 2 above)
Comment (zemlyanoi) [reply src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacade.java]: fixed

**Pull request merged**